File: bigdata/src/main/java/com/hl/bigdata/flink/stream/java/source/MySourceParallelism.java
package com.hl.bigdata.flink.stream.java.source;

import org.apache.flink.configuration.Configuration;
import org.apache.flink.streaming.api.functions.source.RichParallelSourceFunction;

/**
 * @author huanglin
 * @date 2024/03/12 17:20
 */
public class MySourceParallelism extends RichParallelSourceFunction<Integer> {

    private Integer count = 1;
    private boolean isRunning = true;

    @Override
    public void run(SourceContext<Integer> sourceContext) throws Exception {
        while(isRunning) {
            sourceContext.collect(count);
            count++;
            Thread.sleep(1000);
        }
    }

    @Override
    public void cancel() {
        isRunning = false;
    }

    @Override
    public void open(Configuration parameters) throws Exception {
        System.out.println("open multiple source ......");
        super.open(parameters);
    }
}

File: bigdata/src/main/java/com/hl/bigdata/flink/stream/scala/source/MySourceParallelism.scala
package com.hl.bigdata.flink.stream.scala.source

import org.apache.flink.configuration.Configuration
import org.apache.flink.streaming.api.functions.source.{RichParallelSourceFunction, SourceFunction}

/**
 * @author huanglin
 * @date 2024/03/12 17:34
 */
class MySourceParallelism extends RichParallelSourceFunction[Int]{

  var count: Int = 0
  var isRunning: Boolean = true

  override def run(sourceContext: SourceFunction.SourceContext[Int]): Unit = {
    while(isRunning) {
      sourceContext.collect(count)
      count = count + 1
      Thread.sleep(1000)
    }
  }

  override def cancel(): Unit = isRunning = false

  override def open(parameters: Configuration): Unit = {
    println("open multiple source")
    super.open(parameters)
  }
}
